### PR TITLE
Alerting: misc fixes

### PR DESCRIPTION
**Pull request by domasx2**
Miscellaneous small fixes and improvements related to alert list & editing:

* Align DTOs with API changes so saving labels & annotations works again for grafana rules
* Fix bug where blank values would be saved for grafana rule error states
* In rule details view, show datasource list for grafana managed rules same as cloud rules
* Grafana rules now can be filtered by datasource
* "Edit" and "Permissions" icons for grafana rule groups now link to appropriate folder editing pages
* Immediately refetch prom rules state after deleting a rule, so rule is gone immediately from view for grafana managed

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/alerting/unified/components/rules/RulesGroup.tsx [hunk 1]
```diff
@@ -60,11 +61,28 @@ export const RulesGroup: FC<Props> = React.memo(({ group, namespace }) => {
...
-    actionIcons.push(<ActionIcon key="edit" icon="pen" tooltip="edit" />);
-  }
+
+  // for grafana, link to folder views
   if (rulesSource === GRAFANA_RULES_SOURCE_NAME) {
-    actionIcons.push(<ActionIcon key="manage-perms" icon="lock" tooltip="manage permissions" />);
+    const rulerRule = group.rules[0]?.rulerRule;
+    const folderUID = rulerRule && isGrafanaRulerRule(rulerRule) && rulerRule.grafana_alert.namespace_uid;
+    if (folderUID) {
+      const baseUrl = `/dashboards/f/${folderUID}/${kbn.slugifyForUrl(namespace.name)}`;
```
Comment (domasx2) [inline public/app/features/alerting/unified/components/rules/RulesGroup.tsx:70]: is this ok? would it somehow be better to fetch folder url from backend, or ask backend team to include it with rules response?

File: public/app/features/alerting/unified/components/rules/RuleDetails.tsx [hunk 2]
```diff
@@ -23,6 +24,23 @@ export const RuleDetails: FC<Props> = ({ rule, rulesSource }) => {
 
   const annotations = Object.entries(rule.annotations);
 
+  const datasources: Array<{ name: string; icon?: string }> = useMemo(() => {
```
Comment (nathanrodman) [inline public/app/features/alerting/unified/components/rules/RuleDetails.tsx:27]: datasources for threshold alerts, nice 😎

**Review by nathanrodman — APPROVED** [review]
Looks great 🚢

File: public/app/features/alerting/unified/components/rules/RulesGroup.tsx (hunk 1 above)
Comment (peterholmberg) [reply public/app/features/alerting/unified/components/rules/RulesGroup.tsx:70]: You might have to include `config.appSubUrl` like so:
```suggestion
      const baseUrl = `${config.appSubUrl ?? ''}/dashboards/f/${folderUID}/${kbn.slugifyForUrl(namespace.name)}`;
```

Comment (domasx2) [reply public/app/features/alerting/unified/components/rules/RulesGroup.tsx:70]: thanks! I'll go over and see if it's missing for more links as well

- [ ] include `config.appSubUrl`

Comment (domasx2) [reply public/app/features/alerting/unified/components/rules/RulesGroup.tsx]: updated all links to include `appSubUrl`

**Review by peterholmberg — APPROVED** [review]

**Pull request merged**